src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarBukkitEventListener.java

```diff
@@ -244,6 +244,9 @@ public void onPlayerQuit(PlayerQuitEvent event) {
 		if(event.getPlayer().getHealth() > 0
 				&& SiegeWarSettings.getKillPlayersWhoLogoutInSiegeZones()
 				&& SiegeWarDistanceUtil.isLocationInActiveSiegeZone(event.getPlayer().getLocation())) {
+			Town playerTown = TownyAPI.getInstance().getTown(event.getPlayer().getLocation());

```

Comment:
This needs to test that the playerTown isn't null (which happens if they're stood in the wilderness,) and that playerTown.hasResident(event.getPlayer()).

Reply:
Thanks for the quick review :P will push a commit right now to fix

Reply:
This is closer, but it now doesn't handle log-outs that are happening in the wilderness.

`if (playerTown == null || (playerTown != null && playerTown.hasResident(event.getPlayer()))) {`
should probably fix that.

Also, the title of the PR is not quite accurate. You're only preventing killing a player logging off, not preventing  siege sickness.

Reply:
Sorry mate, had been working on two separate PRs and must've named the opposite ones. Will also push your suggested change. Thanks for the help.

Reply:
Reviewing my own code here, might it be more sensical to modify the isLocationInActiveSiegeZone method in the SiegeWarDistanceUtil class. That way, towns near a Siege zone that are not involved in the Siege in any way can be considered exempt from the Siege zone.

Reply:
That might make more sense indeed.